modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridRunningQueryInfo.java

```diff
@@ -169,4 +175,11 @@ public boolean local() {
     public UUID nodeId() {
         return nodeId;
     }
+
+    /**
+     * @return Trace of the current query execution.
+     */
+    public Span span() {
```

Comment:
To be honest I don't like an idea of mixing business code and tracing logic with such methods. However, at least at first glance, it's not obvious how to fix it without dramatic changes in MTC.